src/transformers/models/egt/modeling_egt.py

```diff
@@ -0,0 +1,270 @@
+""" PyTorch EGT model."""
+
+from typing import Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
+
+from ...modeling_outputs import (
+    BaseModelOutputWithNoAttention,
+    SequenceClassifierOutput,
+)
+from ...modeling_utils import PreTrainedModel
+from ...utils import is_dgl_available, requires_backends
+from .configuration_egt import EGTConfig
+
+
+if is_dgl_available():
+    from dgl.nn import EGTLayer
+
+
+NODE_FEATURES_OFFSET = 128
+EDGE_FEATURES_OFFSET = 8
+
+
+_CHECKPOINT_FOR_DOC = "dgl-egt"
+_CONFIG_FOR_DOC = "EGTConfig"
+
+
+EGT_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "Zhiteng/dgl-egt",
+]
+
+
+class VirtualNodes(nn.Module):
+    """
+    Generate node and edge features for virtual nodes in the graph and pad the corresponding matrices.
+    """
+
+    def __init__(self, feat_size, edge_feat_size, num_virtual_nodes=1):
```

Comment:
You need docstrings everywhere to explain what the arguments are, their types, and what each function does